fleetd startup crash when giving weird args

```
~/downloads/fleet-v0.11.5-linux-amd64  
% ./fleetd fdsjkfd
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4020b2]

goroutine 1 [running]:
main.main()
	/home/mischief/code/go/src/github.com/coreos/fleet/fleet/gopath/src/github.com/coreos/fleet/fleetd/fleetd.go:55 +0x14b2

goroutine 2 [runnable]:
runtime.forcegchelper()
	/home/mischief/go/src/runtime/proc.go:90
runtime.goexit()
	/home/mischief/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 3 [runnable]:
runtime.bgsweep()
	/home/mischief/go/src/runtime/mgc0.go:82
runtime.goexit()
	/home/mischief/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 4 [runnable]:
runtime.runfinq()
	/home/mischief/go/src/runtime/malloc.go:712
runtime.goexit()
	/home/mischief/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 5 [runnable]:
os/signal.loop()
	/home/mischief/go/src/os/signal/signal_unix.go:19
created by os/signal.init·1
	/home/mischief/go/src/os/signal/signal_unix.go:27 +0x35
```

I'm getting the same thing with a slightly different message.

```sh
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x40245e]

goroutine 1 [running]:
main.main()
	/build/amd64-usr/var/tmp/portage/app-admin/fleet-0.11.5-r1/work/gopath/src/github.com/coreos/fleet/fleetd/fleetd.go:55 +0x145e

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/lib/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [runnable]:
os/signal.loop()
	/usr/lib/go/src/os/signal/signal_unix.go:20
created by os/signal.init.1
	/usr/lib/go/src/os/signal/signal_unix.go:28 +0x37
```

--- a/fleetd/fleetd.go
+++ b/fleetd/fleetd.go
@@ -42,8 +42,13 @@ func main() {
 	printVersion := userset.Bool("version", false, "Print the version and exit")
 	cfgPath := userset.String("config", "", fmt.Sprintf("Path to config file. Fleet will look for a config at %s by default.", DefaultConfigFile))
 
+	userset.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
+		userset.PrintDefaults()
+	}
+
 	err := userset.Parse(os.Args[1:])
-	if err == flag.ErrHelp {
+	if err != nil {
 		userset.Usage()
 		os.Exit(1)
 	}

